Gen4: move insert planner to gen4 (#12934)
vitessio/vitess: changelog/17.0/17.0.0/summary.md

    - [Support for the `noblob` binlog row image mode](#noblob)
  - **[VTGate](#vtgate)
    - [StreamExecute GRPC API](#stream-execute)

>     - [Insert Planner Gen4] (#insert-planner)

Reviewer: ```suggestion
    - [Insert Planner Gen4](#insert-planner)
```

This does not impact anyone using the mysql client library to connect to VTGate.
This could be a breaking change for grpc api users based on how they have implemented their grpc clients.

#### <a id="insert-planner"/> Insert Planning with Gen4

> Gen4 planner was made default in v14 for select queries. In v15 update and delete queries were moved to Gen4 framework.
> With this release Insert planning is moved to Gen4. 

Reviewer: ```suggestion
Gen4 planner was made default in v14 for `SELECT` queries. In v15 `UPDATE` and `DELETE` queries were moved to Gen4 framework.
With this release `INSERT` queries are moved to Gen4. 
```
